logprep/connector/file/input.py

```diff
@@ -0,0 +1,346 @@
+            file.readlines()
+            initial_filepointer = file.tell()
+        return initial_filepointer
+
+    def _get_initial_file_offset(self, file_name: str) -> int:
+        """Calculates the file_size for given logfile if it's configured to start
+        at the end of a log file"""
+        initial_filepointer = 0
+        if self._config.start == "begin":
+            initial_filepointer = 0
+        elif self._config.start == "end":
+            initial_filepointer = self._get_file_size(file_name)
+
+        return initial_filepointer
+
+    @threadsafe_function
+    def _follow_file(self, file_name: str):
+        """Put log_line as a dict to threadsafe message queue from given input file.
+        Depending on configuration it will continuously monitor a given file for new
+        appending log lines. Depending on configuration it will start to process the
+        given file from the beginning or the end. Will create and continously check
+        the file fingerprints to detect file changes that typically occur on log rotation."""
+
+        with open(file_name, encoding="utf-8") as file:
+            # creating the baseline fingerprint for the file
+            if not self._fileinfo_util.get_fingerprint(file_name):
+                baseline_crc32, baseline_fingerprint_size = self._calc_file_fingerprint(file)
+                self._fileinfo_util.add_fingerprint(
+                    file_name, baseline_crc32, baseline_fingerprint_size
+                )
+
+            baseline_fingerprint_size = self._fileinfo_util.get_fingerprint_size(file_name)
+            check_crc32, _ = self._calc_file_fingerprint(file, baseline_fingerprint_size)
+            if not self._fileinfo_util.check_fingerprint(file_name, check_crc32):
+                # if the fingerprint in check_crc32 is not the same a non-appending
+                # file change happened, create new baseline with potential new fingerprint_size
+                new_check_crc32, new_fingerprint_size = self._calc_file_fingerprint(file)
+                self._fileinfo_util.add_fingerprint(
+                    file_name, new_check_crc32, new_fingerprint_size
+                )
+                self._fileinfo_util.add_offset(file_name, 0)
+
+            while True:
+                if self._fileinfo_util.get_offset(file_name):
+                    # if the file offset already exists, continue from there
+                    file.seek(self._fileinfo_util.get_offset(file_name))
+
+                line = file.readline()
+                # add new file offset after line of file was read
+                self._fileinfo_util.add_offset(file_name, file.tell())
+                if line != "":
+                    if line.endswith("\n"):
+                        message = self._line_to_dict(line)
+                        if message:
+                            self._messages.put(message)
+                else:
+                    break
```

Comment:
Consider refactoring this to tiny little functions with good descriptive names and delete the comments.